### ISSUE TITLE
Edit widget fails to activate under certain conditions.

### ISSUE THREAD

**Issue by mrjrieke** [fyne-io/fyne#1930]
If you create an empty entry (no initial text) that is initially Hidden, and later activate it with Show(), the Widget refuses to display any and all inputs.  This is kind of a corner case, but not necessarily.  Note, it could also be that I'm initializing the widget incorrectly, so please feel free to indicate so and why and close this if it is.

Steps to reproduce the behaviour:
1. Create initial entry with no default content.
2. Hide this entry.
3. Sometime later, do entry.Show() to begin accepting input.
4. Type in some runes.  Notice that no runes display.

Expected Result:
Text would display as it is entered.

Actual Result: 
None of the entered text is displayed.

Example code:
entry := widget.NewEntry()
entry.Resize(entry.MinSize())
entry.Hide()

later...
entry.Show()
... enter runes.

Work around:
1. Call entry.SetText("Some text")
For some reason this works fine!

### Device (please complete the following information):
OS: Linux
Version: Debian Buster
Go version: 1.15.8
Fyne version: 2.0

**Comment by andydotxyz** [comment]
The `entry.Resize(entry.MinSize())` line should not be needed.
In the steps to reproduce this I think it was not clear how / when the element would be focused (by the code or by the user?).

**Comment by mrjrieke** [comment]
Yes, entry.Resize has no effect.  I think focus is driven by user when they Tap on the entry.  That part seems to work as I am able to type text and see it appear in the widget (after the hack).

**Comment by andydotxyz** [comment]
I cannot replicate, I tried the following code and it works as expected:
```go
func main() {
	a := app.New()
	w := a.NewWindow("Hello")

	hello := widget.NewEntry()
	hello.Hide()
	w.SetContent(container.NewVBox(
		hello,
		widget.NewButton("Hi!", func() {
			hello.SetText("Welcome :)")
		}),
	))

	go func() {
		time.Sleep(time.Second)
		hello.Show()
		w.Content().Refresh()
	}()

	w.ShowAndRun()
}
```

Can you see what is different so we can figure what is not working for your code?

**Comment by andydotxyz** [comment]
Closing due to lack of activity
